From ecashin@coraid.com Tue Feb  7 08:45:58 2006
Message-ID: <80311f892d2d2a1267fff540f00786e5@coraid.com>
From: "Ed L. Cashin" <ecashin@coraid.com>
Date: Tue, 7 Feb 2006 11:37:24 -0500
Cc: ecashin@coraid.com, Greg K-H <greg@kroah.com>
Subject: aoe [2/3]: don't request ATA device ID on ATA error

On an ATA error response, take the device down instead of
sending another ATA device identify command.

Signed-off-by: "Ed L. Cashin" <ecashin@coraid.com>


---
 drivers/block/aoe/aoecmd.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/block/aoe/aoecmd.c
+++ gregkh-2.6/drivers/block/aoe/aoecmd.c
@@ -517,6 +517,8 @@ aoecmd_ata_rsp(struct sk_buff *skb)
 	ahout = (struct aoe_atahdr *) (f->data + sizeof(struct aoe_hdr));
 	buf = f->buf;
 
+	if (ahout->cmdstat == WIN_IDENTIFY)
+		d->flags &= ~DEVFL_PAUSE;
 	if (ahin->cmdstat & 0xa9) {	/* these bits cleared on success */
 		printk(KERN_CRIT "aoe: aoecmd_ata_rsp: ata error cmd=%2.2Xh "
 			"stat=%2.2Xh from e%ld.%ld\n", 
@@ -549,7 +551,6 @@ aoecmd_ata_rsp(struct sk_buff *skb)
 				return;
 			}
 			ataid_complete(d, (char *) (ahin+1));
-			d->flags &= ~DEVFL_PAUSE;
 			break;
 		default:
 			printk(KERN_INFO "aoe: aoecmd_ata_rsp: unrecognized "
